## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#230b7472-eab7-4b53-8e1e-aed8cb4d9311) |
| mint-compression-dist-xl.sh | :x: [more...](#a2e22240-92aa-460c-bdeb-7b5c7de50951) |
| mint-compression-fs.sh | :x: [more...](#15e576ef-e51d-4d78-b468-ebbbb388beb4) |
| mint-dist-xl.sh | :x: [more...](#0377b459-cc74-4aaa-a8ad-7cca73613797) |
<a name="a2e22240-92aa-460c-bdeb-7b5c7de50951"></a>
### 7477-b4627b3/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:32319
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3b92b0cd906d:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-emzuh531evb9ed",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 2354,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:32319/aws-sdk-go-test-emzuh531evb9ed: dial tcp 72.28.97.57:32319: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="230b7472-eab7-4b53-8e1e-aed8cb4d9311"></a>
### 7477-b4627b3/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:31577
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 73e49f988cb8:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 56 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 12 seconds
(8/14) Running minio-dotnet tests ... done in 30 seconds
(9/14) Running minio-go tests ... done in 48 seconds
(10/14) Running minio-java tests ... done in 2 minutes and 29 seconds
(11/14) Running minio-js tests ... done in 38 seconds
(12/14) Running minio-py tests ... FAILED in 4 minutes and 54 seconds
{
  "message": "AccessDenied: message: Access Denied",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "duration": 105,
  "args": {
    "object_name": "1eb7c607-1b77-4949-8572-f555291bf8d0",
    "bucket_name": "minio-py-test-8eef960a-a037-46ef-8b09-dc4b341e29e4"
  },
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "status": "FAIL"
}

Executed 11 out of 14 tests successfully.
```
<a name="0377b459-cc74-4aaa-a8ad-7cca73613797"></a>
### 7477-b4627b3/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:31559
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8ca66ae6cc0c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6ja14mvybnwf6m",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 2429,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:31559/aws-sdk-go-test-6ja14mvybnwf6m: dial tcp 72.28.97.57:31559: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="15e576ef-e51d-4d78-b468-ebbbb388beb4"></a>
### 7477-b4627b3/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:30919
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp cc0329c6a1c7:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-lhikdcrchh0vv5",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3065,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:30919/aws-sdk-go-test-lhikdcrchh0vv5: dial tcp 72.28.97.57:30919: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,6 +18,14 @@ matrix:
         - ARCH=x86_64
         - CGO_ENABLED=0
         - GO111MODULE=on
+      # Enable build cache
+      # https://restic.net/blog/2018-09-02/travis-build-cache
+      cache:
+       directories:
+         - $HOME/.cache/go-build
+         - $HOME/gopath/pkg/mod
+         - $HOME/go/pkg/mod
+
       go: 1.12.1
       script:
         - make
@@ -37,7 +45,6 @@ matrix:
       go: 1.12.1
       script:
         - go build --ldflags="$(go run buildscripts/gen-ldflags.go)" -o %GOPATH%\bin\minio.exe
-        - for d in $(go list ./... | grep -v browser); do CGO_ENABLED=1 go test -v -race --timeout 20m "$d"; done
         - bash buildscripts/go-coverage.sh
 
 before_script:

